src/utilities/clipboard-utils.ts

```diff
@@ -1,21 +1,47 @@
 import { gImageMap, ImageMapEntryType } from "../models/image-map";
 
 interface IOnCompleteParams {
-  file: File;
   image: ImageMapEntryType;
 }
 type OnComplete = (params: IOnCompleteParams) => void;
 
-export const pasteClipboardImage = async (onComplete: OnComplete) => {
-  const clipboardContents = await navigator.clipboard.read();
-  if (clipboardContents.length > 0) {
-    if (clipboardContents[0].types.includes("image/png")) {
-      clipboardContents[0].getType("image/png").then(blob => {
-        const blobToFile = new File([blob], "clipboard-image");
-        gImageMap.addFileImage(blobToFile).then(image => {
-          onComplete({ file: blobToFile, image });
-        });
-      });
+export const pasteClipboardImage = async (imageData: any, onComplete: OnComplete) => {
+  if (imageData.image) {
+    const blobToFile = new File([imageData.image], "clipboard-image.png");
+    gImageMap.addFileImage(blobToFile).then(image => {
+      onComplete({ image });
+    });
+  } else if (imageData.text) {
+    const url = imageData.text.match(/curriculum\/([^/]+\/images\/.*)/);
+    if (!url) {
+      console.error("ERROR: invalid image URL");
+      return;
     }
+    const fileUrl = url[1];
+    const filename = fileUrl.split("/").pop();
+    const imageEntry = await gImageMap.getImage(fileUrl, {filename});
+    onComplete({ image: imageEntry });
+  } else {
+    console.error("ERROR: unknown clipboard content type");
```

Comment:
It'd be more useful if the unknown types were part of the message to give us a clue about why the error happened. I guess to do that `getClipboardContent` should include a field that is an array of all of the types that were found on the clipboard.